makefile, touch, file class, circleci config update

lebbenszki opened the pull request:
Made a makefile, edited CircleCi config file, working touch command.

lebbenszki committed:
makefile, touch, file class

lebbenszki committed:
makefile

hegyhati reviewed (changes requested):
@lebbenszki a CI configba keruljon bele olyan step is, ami futtatja az elkeszult binarist es i/o teszteli 2-3 peldan.

hegyhati commented:
@wajzy a CI cofig / make dolgot nezem en (make rendben amugy), magat a kodot kerlek valamikor vizslasd meg.

wajzy commented on directory.cpp:
```diff
@@ -7,15 +7,18 @@ string directory::getName() const
 
 void directory::print()
```
Ez is lehetne const tagfüggvény, de ennyi erővel ezt az egy sor kódot át is lehetne tenni az ls-be, vagy fordítva

wajzy commented on directory.h:
```diff
@@ -12,19 +13,23 @@ class directory
     string name;
     directory* parent;
     list <directory*> subdirectories;
```
Megfontolnám a map használatát, ha a bejegyzések nevétől elvárjuk, hogy egyedi legyen

wajzy commented on directory.h:
```diff
@@ -12,19 +13,23 @@ class directory
     string name;
     directory* parent;
     list <directory*> subdirectories;
+    list <file*> files;
```
Lehet azonos nevű fájl és mappa egy mappában a külön tárolók miatt

wajzy commented on directory.h:
```diff
@@ -12,19 +13,23 @@ class directory
     directory *getParent() const;
     bool hasDirs();
     void rm(string);
     void segedrmrf();
```
Ennek feltétlenül nyilvánosnak kell lennie?

wajzy commented on filesystem.cpp:
```diff
@@ -24,38 +24,60 @@ int filesystem::mkdir(string n)
     return 1;
 }
-
+int filesystem::touch(string filename){
```
A függvény nem ad hibaüzeneteket.

wajzy commented on filesystem.cpp:
```diff
@@ -24,38 +24,60 @@ int filesystem::mkdir(string n)
     directory* tmp;
     tmp=currentdir;
-    list<string> path;
+    path.push_front(currentdir->getName());
```
Miért kell, hogy a path adattag legyen? Miért nem lehet lokális változó?

wajzy commented on filesystem.cpp:
```diff
@@ -24,38 +24,60 @@ int filesystem::mkdir(string n)
         return 1;
         }
     }
+    if(tmp==0){
```
Nem értem, minek kell a tmp. Ha értéke 1, akkor el sem jut ide a vezérlés a fenti return miatt.

wajzy commented on filesystem.cpp:
```diff
@@ -113,8 +135,14 @@ void filesystem::start(){
+                   this->touch(argument1);
+               }
+           }
            else std::cout<<"Unknown command"<<std::endl;
```
Ez a quit-re is kiírja, hogy Unknown command. De egyébként miért nem exit?

wajzy commented on filesystem.h:
```diff
@@ -9,10 +9,12 @@ class filesystem
 private:
     list <directory*> directories;
```
Ez mihez kell?

wajzy reviewed (changes requested):
Mivel a fájloknak és a könyvtáraknak vannak közös tulajdonságaik (név, szülő könyvtár) és ezekhez kapcsolódó funkcióik, javaslom egy közös ősből örököltetni ezeket a közös képességeket.

lebbenszki committed:
patch1.0

circleci config majd csak egy másik branch en lesz átírva mert a tesztben az új parancsokat is teszteli ami ebben a verzióvan még nincs megírva.

lebbenszki committed:
Patch1.1

Makefile lemaradt🤦‍♀️🤦‍♀️🤦‍♀️

lebbenszki commented:
@hegyhati CircleCi config nem let átírva mert a többi parancsal együtt írtuk meg a tesztet csak feldaraboltuk hogy ne egy branchbe legyen feltöltve az egész.

hegyhati commented:
@wajzy kerlek nezd meg, hogy mindent megcsinaltak-e, amit kellett.

@lebbenszki Azert ne uljetek forditva a lovon. Ne csinaljatok meg egyben, aztan szedjetek szet branchekre, annak semmi ertelme. Direkt azert vannak kulon branchek, hogy azok egymastol fuggetlenul menjenek, ahogy a valosagban is tortenne, Es ha jonnek elo conflictok,azokat fel kelljen oldani.
